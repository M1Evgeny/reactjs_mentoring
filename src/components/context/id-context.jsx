import React from 'react';

const IdContext = React.createContext();

export function IdProvider(props) {
  const [movieId, setMovieId] = React.useState(0);
  const value = React.useMemo(() => ({ movieId, setMovieId }), [movieId]);
  return <IdContext.Provider value={value} {...props} />
}

export function useId() {
  const context = React.useContext(IdContext);
  if (!context) {
    throw new Error(`useId must be used within a IdProvider`);
  }
  return context;
}
