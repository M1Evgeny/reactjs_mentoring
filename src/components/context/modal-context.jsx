import React from 'react';

const ModalContext = React.createContext();

export function ModalProvider(props) {
    const [modalObject, setModalObject] = React.useState({});
    const value = React.useMemo(() => ({ modalObject, setModalObject }), [modalObject]);
    return <ModalContext.Provider value={value} {...props} />;
}

export function useModal() {
    const context = React.useContext(ModalContext);
    if (!context) {
      throw new Error(`useModal must be used within a ModalContext`);
    }
    return context; 
}
