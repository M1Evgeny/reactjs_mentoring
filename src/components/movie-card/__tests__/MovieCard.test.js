import React from "react";
import { screen, act, render } from "@testing-library/react";
import userEvent from "@testing-library/user-event";
import { Router } from "react-router-dom";
import { MovieCard } from "..";
import stubs from "../../movie-list-container/mockedMovies.json";
import { ModalProvider } from "../../context/modal-context";
import { createMemoryHistory } from "history";

const mockOnClick = jest.fn();

it("Current render", async () => {
  const history = createMemoryHistory();
  await act(async () => {
    render(
      <Router history={history}>
        <ModalProvider>
          <MovieCard
            id={stubs[0].id}
            title={stubs[0].title}
            genres={stubs[0].genres}
            thumbnail={stubs[0].poster_path}
            releaseDate={stubs[0].release_date}
            onClick={mockOnClick}
          />
        </ModalProvider>
      </Router>
    );
  });
  expect(document.body).toMatchSnapshot();
});

it("Current render", async () => {
  const history = createMemoryHistory();
  await act(async () => {
    render(
      <Router history={history}>
        <ModalProvider>
          <MovieCard
            id={stubs[0].id}
            title={stubs[0].title}
            genres={stubs[0].genres}
            thumbnail={stubs[0].poster_path}
            releaseDate={stubs[0].release_date}
            onClick={mockOnClick}
          />
        </ModalProvider>
      </Router>
    );
  });

  const title = screen.getByText(stubs[0].title);
  expect(title).toBeInTheDocument();

  userEvent.click(title);
  expect(history.length).toBe(2);
});

it("handle edit modal open render", async () => {
  const history = createMemoryHistory();
  await act(async () => {
    render(
      <Router history={history}>
        <ModalProvider>
          <MovieCard
            id={stubs[0].id}
            title={stubs[0].title}
            genres={stubs[0].genres}
            thumbnail={stubs[0].poster_path}
            releaseDate={stubs[0].release_date}
            onClick={mockOnClick}
          />
        </ModalProvider>
      </Router>
    );
  });

  const editButton = screen.getByText("Edit");
  expect(editButton).toBeInTheDocument();

  userEvent.click(editButton);
  //screen.debug();
  // const editModal = screen.getByText('EDIT MOVIE');
  // expect(editModal).toBeInTheDocument();
});

it("handle delete modal open render", async () => {
    const history = createMemoryHistory();
    await act(async () => {
      render(
        <Router history={history}>
          <ModalProvider>
            <MovieCard
              id={stubs[0].id}
              title={stubs[0].title}
              genres={stubs[0].genres}
              thumbnail={stubs[0].poster_path}
              releaseDate={stubs[0].release_date}
              onClick={mockOnClick}
            />
          </ModalProvider>
        </Router>
      );
    });
  
    const deleteButton = screen.getByText("Delete");
    expect(deleteButton).toBeInTheDocument();
  
    userEvent.click(deleteButton);
  });
  
