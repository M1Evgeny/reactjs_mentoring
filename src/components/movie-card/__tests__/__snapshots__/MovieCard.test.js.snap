// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Current render 1`] = `
<body>
  <div>
    <article
      class="filmCard movie-card"
    >
      <div
        class="hamburger_menu"
      >
        <input
          class="menu__toggle"
          id="menu__toggle337167"
          type="checkbox"
        />
        <label
          class="menu__btn"
          for="menu__toggle337167"
        >
          <span />
        </label>
        <ul
          class="menu__box"
        >
          <li>
            <div
              class="menu__item"
              href="#"
              id="edit-button"
            >
              Edit
            </div>
          </li>
          <li>
            <div
              class="menu__item"
              href="#"
              id="delete-button"
            >
              Delete
            </div>
          </li>
        </ul>
      </div>
      <img
        alt="..."
        class="card_img_top"
      />
      <div
        class="card_body"
      >
        <span
          class="filmTitle movie-card-title"
        >
          Fifty Shades Freed
        </span>
        <span
          class="filmYear"
        />
        <p
          class="card-text"
        >
          Drama, Romance
        </p>
      </div>
    </article>
  </div>
</body>
`;
